Sources/MessagingPush/UserNotificationsFramework/UserNotificationsFrameworkAdapter.swift

```diff
@@ -0,0 +1,127 @@
+import CioInternalCommon
+import Foundation
+import UserNotifications
+
+/**
+ The iOS framework, `UserNotifications`, is abstracted away from the SDK codebase.
+
+ This file is the connection between our SDK and `UserNotifications`.
+ In production, iOS will call functions in this file. Those requests are then forwarded onto the abstracted code in the SDK to perform all of the logic.
+ */
+@available(iOSApplicationExtension, unavailable)
+protocol UserNotificationsFrameworkAdapter {
+    // A reference to an instance of UNUserNotificationCenterDelegate that we can provide to iOS in production.
+    var delegate: UNUserNotificationCenterDelegate { get }
+
+    func beginListeningNewNotificationCenterDelegateSet()
+
+    // Called when a new `UNUserNotificationCenterDelegate` is set on the host app. Our Swizzling is what calls this function.
+    func newNotificationCenterDelegateSet(_ newDelegate: UNUserNotificationCenterDelegate?)
+}
+
+/**
+ Keep this class small and simple because it is only able to be tested in QA testing. All logic for handling push events should be in the rest of the code base that has automated tests around it.
+ */
+@available(iOSApplicationExtension, unavailable)
+// sourcery: InjectRegisterShared = "UserNotificationsFrameworkAdapter"
+// sourcery: InjectSingleton
```

Comment:
The big call-out in this file is that I did modify it a little from what is inside of `main`. 

Because `DiGraphShared` is always in-memory, even if the SDK is initialized multiple times, we no longer [need the workarounds](https://github.com/customerio/customerio-ios/blob/cdf2f1fc45f1dadf0966a81c907f0ee1cb244f4f/Sources/MessagingPush/UserNotificationsFramework/UserNotificationsFrameworkAdapter.swift#L28-L52) that were previously needed to keep this class always in-memory. 

So, this PR uses the code generation to add this class to the digraph!